### PR TITLE
Added the ability to generate host certificates to tctl.

### PR DESCRIPTION
**Pull request by russjones**
**Purpose**

As covered in #1458, to support the Teleport recording proxy, Teleport administrators need to have the ability to place host certificates issues by the Teleport CA on OpenSSH nodes. This PR adds the ability to generate arbitrary host certificates to `tctl`.

**Implementation**

```bash
$ tctl auth gen \
   --principal=host.example.com \
   --priv-key=ssh_host_rsa_key \
   --pub-key=ssh_host_rsa_key.pub \
   --cert=ssh_host_rsa_key-cert.pub
```

**Related Issues**

Fixes https://github.com/gravitational/teleport/issues/1458

### CODE REVIEW

File: tool/tctl/common/auth_command.go [hunk 1]
```diff
@@ -52,9 +55,11 @@ func (a *AuthCommand) Initialize(app *kingpin.Application, config *service.Confi
 	a.authExport.Flag("compat", "export cerfiticates compatible with specific version of Teleport").StringVar(&a.compatVersion)
 	a.authExport.Flag("type", "certificate type: 'user' or 'host'").StringVar(&a.authType)
 
-	a.authGenerate = auth.Command("gen", "Generate a new SSH keypair").Hidden()
+	a.authGenerate = auth.Command("gen", "Generate a new SSH host certificate")
 	a.authGenerate.Flag("pub-key", "path to the public key").Required().StringVar(&a.genPubPath)
 	a.authGenerate.Flag("priv-key", "path to the private key").Required().StringVar(&a.genPrivPath)
+	a.authGenerate.Flag("cert", "path to certificate").Required().StringVar(&a.genCertPath)
+	a.authGenerate.Flag("principal", "principal name on certificate").Required().StringVar(&a.principal)
```
Comment (klizhentas) [inline tool/tctl/common/auth_command.go]: what if you need multiple principals?

**Comment by knisbet** [comment]
I'm just curious, have you considered doing something like ssh-copy-id, that can ssh to a remote node, and do all the setup / certificate generation / check that everything is configured correctly / make it easier to roll-out on a wide scale?

**Comment by kontsevoy** [comment]
Having looked deeper into this, it appears we have a much better place for this. We have `tctl auth sign` command which does pretty much the same thing, but for users. 

It currently takes a mandatory `--user` flag. 

## Proposal

Lets move this to `tctl auth sign --host` syntax, where:

* `--host` can take comma-separated list, like `--host=luna,luna.example.com,10.5.5.12`
* `--format` flag is always set to "dir"
* name of the generated files is compatible with OpenSSH (it already is) but the base for the name is the first "host" (i.e. the same thing it does for a `--user` user case).

This way we'll have a consistent CLI: 

1. `tctl auth sign` creates signed credentials for both users and hosts, to give them access to a cluster, 2. while `tctl auth export` does the opposite - exports the CA certs, to let users/hosts trust a cluster.

**Comment by klizhentas** [comment]
Ev's proposal looks good to me

**Comment by kontsevoy** [comment]
@knisbet In a proper Teleport deployment there is no need to "ssh into a remote node" because all secrets are managed via a CA, and `tctl` is the CLI for it. These two commands: `tctl auth sign` and `tctl auth export` generate the keys for two-way trust **in a legacy use case** when some nodes aren't upgraded to Teleport. These keys are to be distributed by other means, probably ansible/chef, to old `sshd` noces. Doing advanced ssh-copy-id will likely not work, because there's usually a reason those nodes are "legacy": network appliances or private Github repositories, i.e. placement of the credentials isn't straightforward.

**Comment by russjones** [comment]
I'm going to temporarily close this while I work on it.

**Pull request closed without merging**